suzieq/engines/pandas/bgp.py

```diff
@@ -272,6 +273,10 @@ def _check_if_state(row, if_df):
                 else [], axis=1)
 
             failed_df['result'] = 'fail'
+            failed_df['assertReason'] += failed_df .apply(
+                lambda x: ['Not established']
+                if len(x.assertReason) == 0 else [],
+                axis=1)
```

Comment:
Luca, you're doing this to avoid the assertReason being blank? Doesn't this show up in all failures then, as an additional reason to anything else we might find? If so, I'm not sure this is useful. The state field we display should be sufficient to indicate that this failed, and hence the result is fail. Maybe we should set it to 'unknown cause' or something like that.

Reply:
Yes. This message is going to be displayed only when the assertion is failing with a blank reason. You right, I'll change it to 'Unknow reason'